done/done.py

```diff
@@ -1,37 +1,72 @@
 """ Show a toggle which lets students mark things as done."""
 
-import pkg_resources
 import uuid
 
+import pkg_resources
+from django import utils
 from xblock.core import XBlock
-from xblock.fields import Scope, String, Boolean, DateTime, Float
+from xblock.fields import Boolean, DateTime, Float, Scope, String
 from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+from xblockutils.settings import ThemableXBlockMixin, XBlockWithSettingsMixin
+
+from .utils import DummyTranslationService, _
+
 
 def resource_string(path):
     """Handy helper for getting resources from our kit."""
     data = pkg_resources.resource_string(__name__, path)
     return data.decode("utf8")
 
 
-class DoneXBlock(XBlock):
+@XBlock.wants('settings')
+@XBlock.needs('i18n')
+class DoneXBlock(XBlock, XBlockWithSettingsMixin, ThemableXBlockMixin):
     """
     Show a toggle which lets students mark things as done.
     """
 
     done = Boolean(
         scope=Scope.user_state,
-        help="Is the student done?",
+        help=_("Is the student done?"),
         default=False
     )
 
     align = String(
         scope=Scope.settings,
-        help="Align left/right/center",
-        default="left"
+        help=_("Align left/right/center"),
+        default=_("left")
     )
 
     has_score = True
 
+    loader = ResourceLoader(__name__)
+
+    block_settings_key = 'done'
```

Comment:
Please point me where block_settings_key and default_theme_config are used?

Reply:
This code doesn't use those settings. I deleted it.